shared/src/store/action-reducers/simulation.ts

```diff
@@ -0,0 +1,89 @@
+import type { Mutable } from '../../utils';
+import { UUID, uuidValidationOptions } from '../../utils';
+import { IsValue } from '../../utils/validators';
+import type { Action, ActionReducer } from '../action-reducer';
+import { getElement } from './utils';
+
+export class UpdateTreatPatientsIntervalsAction implements Action {
+    @IsValue('[TreatPatientsBehavior] Update TreatPatientsIntervals' as const)
+    public readonly type =
+        '[TreatPatientsBehavior] Update TreatPatientsIntervals';
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly simulatedRegionId!: UUID;
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly behaviorStateId!: UUID;
+
+    @IsNumber()
+    @Min(-1)
+    public readonly unknown!: number;
+
+    @IsNumber()
+    @Min(-1)
+    public readonly counted!: number;
+
+    @IsNumber()
+    @Min(-1)
+    public readonly triaged!: number;
+
+    @IsNumber()
+    @Min(-1)
+    public readonly secured!: number;
+
+    @IsNumber()
+    @Min(-1)
+    public readonly countingTimePerPatient!: number;
+}
+
+export namespace SimulationActionReducers {
+    export const updateTreatPatientsIntervals: ActionReducer<UpdateTreatPatientsIntervalsAction> =
+        {
+            action: UpdateTreatPatientsIntervalsAction,
+            /*
+             *   unknown, counted, triaged, secured, countingTimePerPatient stay the same when -1
+             */
+            reducer: (
+                draftState,
+                {
+                    simulatedRegionId,
+                    behaviorStateId,
+                    unknown,
+                    counted,
+                    triaged,
+                    secured,
+                    countingTimePerPatient,
+                }
+            ) => {
+                const simulatedRegion = getElement(
+                    draftState,
+                    'simulatedRegion',
+                    simulatedRegionId
+                );
+                const behaviorStates = simulatedRegion.behaviors;
+                const treatPatientsBehaviorState = behaviorStates.find(
+                    (behaviorState) => behaviorState.id === behaviorStateId
+                ) as Mutable<TreatPatientsBehaviorState>;
+                if (unknown >= 0) {
+                    treatPatientsBehaviorState.intervals.unknown = unknown;
+                }
+                if (counted >= 0) {
+                    treatPatientsBehaviorState.intervals.counted = counted;
+                }
+                if (triaged >= 0) {
+                    treatPatientsBehaviorState.intervals.triaged = triaged;
+                }
+                if (secured >= 0) {
+                    treatPatientsBehaviorState.intervals.secured = secured;
+                }
+                if (countingTimePerPatient >= 0) {
+                    treatPatientsBehaviorState.intervals.countingTimePerPatient =
+                        countingTimePerPatient;
+                }
+                return draftState;
```

Comment:
I don't like the `-1` hack, because it is prone to error, unintuitive and undocumented. A few options how to deal with it:

* improve the methods in the frontend to resend unchanged values (! use a helper that takes the status as parameter, don't stuff the template with js code !)
* adapt the action to have a `status` field and set only one field at a time
* make the interval optional for every status.  (my preference if this is easily doable for actions)

Reply:
Every time I formally sent a -1 I will now instead send undefined